File: solutions/0009-palindrome-number/solution.py
"""
LeetCode: [0009] Palindrome Number
Link: https://leetcode.com/problems/palindrome-number/
Date: 2026-06-03
Language: python3
Status: Accepted
"""

class Solution:
    def isPalindrome(self, x: int) -> bool:
        s = str(x)
        sr = s[::-1]
        return s == sr
        

File: solutions/0009-palindrome-number/solution.java
/**
 * LeetCode: [0009] Palindrome Number
 * Link: https://leetcode.com/problems/palindrome-number/
 * Date: 2026-06-05
 * Language: java
 * Status: Accepted
 */

class Solution {
    public boolean isPalindrome(int x) {
        String s = String.valueOf(x);
        int len = s.length();
        for (int i = 0; i <= len / 2; i++) {
            if (s.charAt(i) != s.charAt(len - 1 -i))
                return false;
        }
        return true;
    }
}